### ISSUE TITLE
A compare option. It would be helpful if I could see the differences in size and resolution between photos. This would be used to see which duplicate is the better one to keep. (This is a 'sky's the limit request'. I don't expect it to be added).

### ISSUE THREAD

**Issue by puregallery** [puregallery/Pure-Gallery-Support#6]

**Comment by Nostang3** [comment]
I had a design idea of a way to implement this with the details option. 

In my usage, I'm often having to tag photos new photos added to the album, then filter PG for those tags to find duplicates or similar photos. I then have to compare them to each other to find out which one I want to keep. Usually via comparing size and resolution. Currently, I found that I select the similar files, then choose to apply tags to them, and then use the details button on each photo to compare. This works but is time consuming and i'm opening a closing a lot of dialogs. 

You could tie this in with the details selection in the album view. I can select 2-3 photos and then hit the details option. It would then bring up a dialog with a limited set of details. 

Currently the dialog shows Name, Path, Favorite, Size, Album, MimeType, Dimensions, Date. 
For the compare dialog you would only really need Name, Size and Dimensions because everything else is pretty much the same. Or you can have option in settings to customize what is shown in the comparison.

Possible Example of dialog:
>Name Stupid_long_name.jpg
>Size 4.2MB    Dimensions: 1920x1080

>Name Stupid_long_name_2.jpg
>Size 4.3MB    Dimensions: 1920x1080

>Name Stupid_long_name_3.jpg
>Size 2.2MB    Dimensions: 1920x1080

Or you may create a similar dialog like the Apply Tags dialog but at the bottom you could have the information in the place of Current Tags and the Add Tags/Remove Tags/Fav/Info section. 

I'm torn on which I think would be better personally. The former would be better for quick compare and then be able to unselect what you want to keep then quickly delete the one you don't. Although, the latter gives you the option to clearly see which photo the information is tied too which gives you the option to compare very similar photos and keep one that might be smaller size/dimension but is the better photo. If you could add a delete button the latter method it could be the best of both worlds. 

Long post, I know.

**Comment by puregallery** [comment]
I like the second option too. Right now, to modify the tags, we call it "Edit Tags".
What should we call this operation? Just 'compare' will make sense only to the two of us frankly.

**Comment by Nostang3** [comment]
If you access it via the details option then maybe "Detail Comparison". I mean you might not actually have to give it a name. You are still viewing details, just details of multiple photos. Selecting multiple photos and hitting the details button and seeing information for each photo seems intuitive enough to me not to have to name it.

**Comment by puregallery** [comment]
With the newly added metadata on images, and the ability to open the selected images in viewer...do you need anything else to address this?

**Comment by Nostang3** [comment]
I've been using tag mode to compare photos currently but I still have to hit the info button on a photo, close dialog, swipe to the next photo and hit the info button again. The metadata visible is great if the photos are right by each other, but when I'm comparing a photo from today and one posted a year or two ago, then it becomes hard to go back and forth to compare a few variables. 

Maybe instead of a dialog that pops up on the tag mode, you could instead toggle the visible metadata on the picture. This way you can turn on the info metadata and swipe between photos. Then turn it off when you start tagging. 

I use the compare mode because sometimes a new photos is added to the album that might already be there. It might be a better resolution, or size, or bigger crop. Or it could be smaller. So filter by tags, then select the photos that are similar, then I use the info so I can decide which photo I want to keep so I don't have duplicates.

**Comment by puregallery** [comment]
>The metadata visible is great if the photos are right by each other

This is possible in the app. Select the photos you want to compare and click the left most icon in the bottom bar. That will open the selected images in the viewer next to each other.

>Maybe instead of a dialog that pops up on the tag mode, you could instead toggle the visible metadata on the picture

Either way, I will look into this too...

**Comment by Nostang3** [comment]
`This is possible in the app. Select the photos you want to compare and click the left most icon in the bottom bar. That will open the selected images in the viewer next to each other.`

This is what I do now as described above with 'tag mode'. While they next to each other comparing resolution or size tags multiple clicks and swiping.

**Comment by puregallery** [comment]
>This is what I do now as described above with 'tag mode'. While they next to each other comparing resolution or size tags multiple clicks and swiping.

I'm a bit confused...you meant the tag mode by clicking on the tag icon on the bottom bar right? I'm not sure when I included this feature...but you should be able to open the selected items in 'media viewer' by clicking the left most icon on the bottom bar. Use this feature until a full-fledged compare option is available...this might be more useful to you since you can now edit the tags from 'media viewer' screen also.

**Comment by puregallery** [comment]
Cool...is that helping u in the meantime?
I guess I was sitting on that feature for a long without releasing it...got confused.

**Comment by Nostang3** [comment]
Yeah, it's really helpful for me to compare metadata between duplicate photos. Saves me a bunch of clicks. I think this case can be closed because of the media view feature and the metadata displayed. If you have no objections then you can close it.
